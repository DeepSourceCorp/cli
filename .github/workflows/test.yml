name: Unit Tests

on:
  push:
    branches: [develop]
  pull_request:
    branches: [develop]

jobs:
  test:
    runs-on: ubuntu-latest
    env:
      NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
      ZEAL_SECRET: ${{secrets.GITHUB_TOKEN}}

    steps:
      - name: Checkout Bifrost
        uses: actions/checkout@v2

      - name: Setting Up Node
        uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          always-auth: true
          registry-url: https://npm.pkg.github.com/
          scope: '@deepsourcelabs'
          cache: 'yarn'

      - name: Install packages
        run: yarn

      - name: Run Test
        run: yarn test --coverage=true --coverageReporters=cobertura

      - name: Install DeepSource CLI
        run: curl https://deepsource.io/cli | sh

      - name: Set DEEPSOURCE_DSN
        run: |
          ./bin/deepsource report --analyzer test-coverage --key javascript --value-file ./coverage/cobertura-coverage.xml
        env:
          DEEPSOURCE_DSN: ${{ secrets.BIFROST_DSN }}
