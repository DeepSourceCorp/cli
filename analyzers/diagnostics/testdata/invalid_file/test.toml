name = "2do-Checker"
