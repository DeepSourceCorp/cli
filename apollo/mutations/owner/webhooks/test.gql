mutation testWebhook($webhookId: ID!) {
  testWebhook(input: { webhookId: $webhookId }) {
    ok
  }
}
