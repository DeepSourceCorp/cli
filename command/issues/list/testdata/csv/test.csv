analyzer,issue_code,issue_title,occurence_title,issue_category,path,begin_line,begin_column,end_line,end_column
go,RVV-B0013,Unused method receiver detected,Unused method receiver detected,,deepsource/transformers/queries/get_transformers.go,34,0,34,0
go,RVV-B0013,Unused method receiver detected,Unused method receiver detected,,deepsource/transformers/queries/get_transformers.go,44,0,44,0
