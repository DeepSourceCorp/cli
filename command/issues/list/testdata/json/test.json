{
	"occurences": [
		{
			"analyzer": "go",
			"issue_code": "RVV-B0013",
			"issue_title": "Unused method receiver detected",
			"occurence_title": "Unused method receiver detected",
			"issue_category": "",
			"location": {
				"path": "deepsource/transformers/queries/get_transformers.go",
				"position": {
					"begin": {
						"line": 34,
						"column": 0
					},
					"end": {
						"line": 34,
						"column": 0
					}
				}
			}
		},
		{
			"analyzer": "go",
			"issue_code": "RVV-B0013",
			"issue_title": "Unused method receiver detected",
			"occurence_title": "Unused method receiver detected",
			"issue_category": "",
			"location": {
				"path": "deepsource/transformers/queries/get_transformers.go",
				"position": {
					"begin": {
						"line": 44,
						"column": 0
					},
					"end": {
						"line": 44,
						"column": 0
					}
				}
			}
		}
	],
	"summary": {
		"total_occurences": 2,
		"unique_issues": 1
	}
}