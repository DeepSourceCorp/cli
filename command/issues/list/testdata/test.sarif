{
  "version": "2.1.0",
  "$schema": "https://json.schemastore.org/sarif-2.1.0-rtm.5.json",
  "runs": [
    {
      "tool": {
        "driver": {
          "informationUri": "https://deepsource.io",
          "name": "deepsource",
          "rules": [
            {
              "id": "RVV-B0013",
              "name": "Unused method receiver detected",
              "shortDescription": {
                "text": ""
              },
              "helpUri": "",
              "properties": {
                "category": "",
                "recommended": ""
              }
            }
          ]
        }
      },
      "results": [
        {
          "ruleId": "RVV-B0013",
          "ruleIndex": 0,
          "kind": "fail",
          "level": "error",
          "message": {
            "text": "Unused method receiver detected"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "deepsource/transformers/queries/get_transformers.go"
                },
                "region": {
                  "startLine": 34,
                  "endLine": 34
                }
              }
            }
          ]
        },
        {
          "ruleId": "RVV-B0013",
          "ruleIndex": 1,
          "kind": "fail",
          "level": "error",
          "message": {
            "text": "Unused method receiver detected"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "deepsource/transformers/queries/get_transformers.go"
                },
                "region": {
                  "startLine": 44,
                  "endLine": 44
                }
              }
            }
          ]
        }
      ]
    }
  ]
}
