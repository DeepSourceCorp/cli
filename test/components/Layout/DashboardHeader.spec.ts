import { render } from '@testing-library/vue'
import { createLocalVue } from '@vue/test-utils'
import { dashboardContextGenerator, mocksGenerator, storeModulesGenerator } from '~/test/mocks'
import { RouterLinkStub } from '@vue/test-utils'

import VTooltip from 'v-tooltip'
import { VueConstructor } from 'vue'
import Vuex from 'vuex'

import DashboardHeader from '~/components/Layout/DashboardHeader.vue'

describe('[[ DashboardHeader ]]', () => {
  let localVue: VueConstructor<Vue>
  let storeMock: Record<string, unknown>

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VTooltip)

    storeMock = {
      modules: storeModulesGenerator()
    }
  })

  const stubs = {
    ZAvatar: true,
    ZAvatarGroup: true,
    ZIcon: true,
    ZTag: true,
    NuxtLink: RouterLinkStub,
    InviteMembersModal: true
  }

  test('renders DashboardHeader', () => {
    const { html } = render(
      DashboardHeader,
      {
        mocks: mocksGenerator(),
        stubs,
        store: new Vuex.Store(storeMock)
      },
      (vue) => {
        vue.use(VTooltip)
      }
    )

    expect(html()).toMatchSnapshot()
  })

  test('renders DashboardHeader on Prem', () => {
    const { html } = render(
      DashboardHeader,
      {
        mocks: mocksGenerator({
          $config: { onPrem: true }
        }),
        stubs,
        store: new Vuex.Store(storeMock)
      },
      (vue) => {
        vue.use(VTooltip)
      }
    )

    expect(html()).toMatchSnapshot()
  })

  test('renders DashboardHeader with paid plan', () => {
    const { html } = render(
      DashboardHeader,
      {
        mocks: mocksGenerator(),
        stubs,
        store: new Vuex.Store({
          modules: storeModulesGenerator({
            'user/active': {
              namespaced: true,
              state: {
                viewer: {
                  dashboardContext: [
                    dashboardContextGenerator({
                      subscribed_plan_info: {
                        name: 'Business',
                        slug: 'business'
                      }
                    })
                  ]
                }
              }
            }
          })
        })
      },
      (vue) => {
        vue.use(VTooltip)
      }
    )

    expect(html()).toMatchSnapshot()
  })

  test('renders DashboardHeader with paid plan but no perms', () => {
    const { html } = render(
      DashboardHeader,
      {
        mocks: mocksGenerator({
          $route: {
            name: 'provider-owner-repo',
            params: {
              provider: 'gh',
              owner: 'not-deepsourcelabs',
              repo: 'bifrost'
            }
          }
        }),
        stubs,
        store: new Vuex.Store({
          modules: storeModulesGenerator({
            'user/active': {
              namespaced: true,
              state: {
                viewer: {
                  dashboardContext: [
                    dashboardContextGenerator({
                      subscribed_plan_info: {
                        name: 'Business',
                        slug: 'business'
                      }
                    })
                  ]
                }
              }
            }
          })
        })
      },
      (vue) => {
        vue.use(VTooltip)
      }
    )

    expect(html()).toMatchSnapshot()
  })

  test('renders DashboardHeader with different vcs', () => {
    const { html } = render(
      DashboardHeader,
      {
        mocks: mocksGenerator(),
        stubs,
        store: new Vuex.Store({
          modules: storeModulesGenerator({
            'user/active': {
              namespaced: true,
              state: {
                viewer: {
                  dashboardContext: [
                    dashboardContextGenerator({
                      vcs_provider_display: 'github-enterprise'
                    })
                  ]
                }
              }
            }
          })
        })
      },
      (vue) => {
        vue.use(VTooltip)
      }
    )

    expect(html()).toMatchSnapshot()
  })
})
