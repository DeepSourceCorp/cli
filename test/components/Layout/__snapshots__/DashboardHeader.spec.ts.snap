// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ DashboardHeader ]] renders DashboardHeader 1`] = `
"<div>
  <div class=\\"overflow-x-auto border-b bg-ink-300 border-ink-200\\">
    <div id=\\"header\\" class=\\"flex items-center p-4\\">
      <h2 class=\\"inline-flex items-center space-x-4 font-medium text-vanilla-100\\"><a class=\\"inline-flex items-center flex-shrink-0 space-x-2\\"><a class=\\"inline-block rounded-full flex-shrink-0 bg-ink-100 p-1.5\\"><img src=\\"https://static.deepsource.io/avatars/14ce08c1-6b0d-4502-9a03-576af9a06f9d.png\\" alt=\\"deepsourcelabs\\" class=\\"rounded-full leading-none h-8 w-8 text-sm\\">
            <!----></a> <span class=\\"text-xl font-semibold cursor-pointer lg:text-xl xl:text-2xl\\">deepsourcelabs</span></a> <span class=\\"inline-flex items-center space-x-3\\"><a href=\\"https://github.com/deepsourcelabs\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"flex items-center has-tooltip\\" data-original-title=\\"null\\"><div class=\\"inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 border-2 border-ink-200 p-0.5 text-sm bg-ink-200\\"><svg viewBox=\\"0 0 24 24\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" class=\\"z-icon w-5 h-5 z-icon--github text-current\\"><path d=\\"M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z\\" fill=\\"currentColor\\"></path></svg><!----></div></a> <a><div class=\\"inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 font-semibold leading-none tracking-wider text-center uppercase border border-ink-100 text-vanilla-300 hover:text-vanilla-100 hover:bg-ink-100 py-1.5 px-3 text-sm bg-ink-200 has-tooltip\\" data-original-title=\\"null\\"><svg viewBox=\\"0 0 24 24\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" class=\\"z-icon w-4 h-4 z-icon--star text-current\\"><polygon points=\\"12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2\\"></polygon></svg><span>Upgrade</span>
        <!---->
    </div></a></span></h2>
  </div>
</div>
</div>"
`;

exports[`[[ DashboardHeader ]] renders DashboardHeader on Prem 1`] = `
"<div>
  <div class=\\"overflow-x-auto border-b bg-ink-300 border-ink-200\\">
    <div id=\\"header\\" class=\\"flex items-center p-4\\">
      <h2 class=\\"inline-flex items-center space-x-4 font-medium text-vanilla-100\\"><a class=\\"inline-flex items-center flex-shrink-0 space-x-2\\"><a class=\\"inline-block rounded-full flex-shrink-0 bg-ink-100 p-1.5\\"><img src=\\"https://static.deepsource.io/avatars/14ce08c1-6b0d-4502-9a03-576af9a06f9d.png\\" alt=\\"deepsourcelabs\\" class=\\"rounded-full leading-none h-8 w-8 text-sm\\">
            <!----></a> <span class=\\"text-xl font-semibold cursor-pointer lg:text-xl xl:text-2xl\\">deepsourcelabs</span></a> <span class=\\"inline-flex items-center space-x-3\\"><a href=\\"https://github.com/deepsourcelabs\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"flex items-center has-tooltip\\" data-original-title=\\"null\\"><div class=\\"inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 border-2 border-ink-200 p-0.5 text-sm bg-ink-200\\"><svg viewBox=\\"0 0 24 24\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" class=\\"z-icon w-5 h-5 z-icon--github text-current\\"><path d=\\"M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z\\" fill=\\"currentColor\\"></path></svg><!----></div></a> <!----></span></h2>
    </div>
  </div>
</div>"
`;

exports[`[[ DashboardHeader ]] renders DashboardHeader with different vcs 1`] = `
"<div>
  <div class=\\"overflow-x-auto border-b bg-ink-300 border-ink-200\\">
    <div id=\\"header\\" class=\\"flex items-center p-4\\">
      <h2 class=\\"inline-flex items-center space-x-4 font-medium text-vanilla-100\\"><a class=\\"inline-flex items-center flex-shrink-0 space-x-2\\"><a class=\\"inline-block rounded-full flex-shrink-0 bg-ink-100 p-1.5\\"><img src=\\"https://static.deepsource.io/avatars/14ce08c1-6b0d-4502-9a03-576af9a06f9d.png\\" alt=\\"deepsourcelabs\\" class=\\"rounded-full leading-none h-8 w-8 text-sm\\">
            <!----></a> <span class=\\"text-xl font-semibold cursor-pointer lg:text-xl xl:text-2xl\\">deepsourcelabs</span></a> <span class=\\"inline-flex items-center space-x-3\\"><a href=\\"https://github.com/deepsourcelabs\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"flex items-center has-tooltip\\" data-original-title=\\"null\\"><div class=\\"inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 border-2 border-ink-200 p-0.5 text-sm bg-ink-200\\"><svg viewBox=\\"0 0 24 24\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" class=\\"z-icon w-5 h-5 z-icon--github text-current\\"><path d=\\"M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z\\" fill=\\"currentColor\\"></path></svg><!----></div></a> <a><div class=\\"inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 font-semibold leading-none tracking-wider text-center uppercase border border-ink-100 text-vanilla-300 hover:text-vanilla-100 hover:bg-ink-100 py-1.5 px-3 text-sm bg-ink-200 has-tooltip\\" data-original-title=\\"null\\"><svg viewBox=\\"0 0 24 24\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" class=\\"z-icon w-4 h-4 z-icon--star text-current\\"><polygon points=\\"12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2\\"></polygon></svg><span>Upgrade</span>
        <!---->
    </div></a></span></h2>
  </div>
</div>
</div>"
`;

exports[`[[ DashboardHeader ]] renders DashboardHeader with paid plan 1`] = `
"<div>
  <div class=\\"overflow-x-auto border-b bg-ink-300 border-ink-200\\">
    <div id=\\"header\\" class=\\"flex items-center p-4\\">
      <h2 class=\\"inline-flex items-center space-x-4 font-medium text-vanilla-100\\"><a class=\\"inline-flex items-center flex-shrink-0 space-x-2\\"><a class=\\"inline-block rounded-full flex-shrink-0 bg-ink-100 p-1.5\\"><img src=\\"https://static.deepsource.io/avatars/14ce08c1-6b0d-4502-9a03-576af9a06f9d.png\\" alt=\\"deepsourcelabs\\" class=\\"rounded-full leading-none h-8 w-8 text-sm\\">
            <!----></a> <span class=\\"text-xl font-semibold cursor-pointer lg:text-xl xl:text-2xl\\">deepsourcelabs</span></a> <span class=\\"inline-flex items-center space-x-3\\"><a href=\\"https://github.com/deepsourcelabs\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"flex items-center has-tooltip\\" data-original-title=\\"null\\"><div class=\\"inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 border-2 border-ink-200 p-0.5 text-sm bg-ink-200\\"><svg viewBox=\\"0 0 24 24\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" class=\\"z-icon w-5 h-5 z-icon--github text-current\\"><path d=\\"M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z\\" fill=\\"currentColor\\"></path></svg><!----></div></a> <a><div class=\\"inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 leading-none text-center border border-ink-100 py-1.5 px-3 text-sm bg-ink-200 has-tooltip\\" data-original-title=\\"null\\"><!---->Business<!----></div></a></span></h2>
    </div>
  </div>
</div>"
`;

exports[`[[ DashboardHeader ]] renders DashboardHeader with paid plan but no perms 1`] = `
"<div>
  <div class=\\"overflow-x-auto border-b bg-ink-300 border-ink-200\\">
    <div id=\\"header\\" class=\\"flex items-center p-4\\">
      <h2 class=\\"inline-flex items-center space-x-4 font-medium text-vanilla-100\\"><a class=\\"inline-flex items-center flex-shrink-0 space-x-2\\"><a class=\\"inline-block rounded-full flex-shrink-0 bg-ink-100 p-1.5\\"><img src=\\"https://static.deepsource.io/avatars/14ce08c1-6b0d-4502-9a03-576af9a06f9d.png\\" alt=\\"deepsourcelabs\\" class=\\"rounded-full leading-none h-8 w-8 text-sm\\">
            <!----></a> <span class=\\"text-xl font-semibold cursor-pointer lg:text-xl xl:text-2xl\\">deepsourcelabs</span></a> <span class=\\"inline-flex items-center space-x-3\\"><a href=\\"https://github.com/deepsourcelabs\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"flex items-center has-tooltip\\" data-original-title=\\"null\\"><div class=\\"inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 border-2 border-ink-200 p-0.5 text-sm bg-ink-200\\"><svg viewBox=\\"0 0 24 24\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" class=\\"z-icon w-5 h-5 z-icon--github text-current\\"><path d=\\"M12 1C9.38799 1.00011 6.8613 1.92962 4.87196 3.62221C2.88263 5.31481 1.56048 7.66004 1.14208 10.2383C0.723674 12.8165 1.23633 15.4595 2.58831 17.6943C3.94029 19.9291 6.04337 21.61 8.52125 22.4361C9.07124 22.5324 9.27749 22.2024 9.27749 21.9136C9.27749 21.6524 9.26373 20.7861 9.26373 19.8649C6.49999 20.3736 5.785 19.1912 5.565 18.5724C5.32088 17.9706 4.93391 17.4373 4.43751 17.0186C4.05251 16.8124 3.50251 16.3036 4.42375 16.2899C4.77551 16.3281 5.11289 16.4505 5.40729 16.6468C5.7017 16.843 5.94446 17.1074 6.11499 17.4174C6.26544 17.6876 6.46774 17.9256 6.7103 18.1175C6.95286 18.3095 7.23091 18.4517 7.52852 18.536C7.82613 18.6203 8.13745 18.645 8.44464 18.6088C8.75184 18.5726 9.04886 18.4761 9.3187 18.3249C9.36632 17.7657 9.61554 17.2428 10.02 16.8537C7.57248 16.5787 5.015 15.6299 5.015 11.4225C4.99954 10.3292 5.40295 9.27151 6.14249 8.46623C5.80621 7.51608 5.84555 6.47336 6.25249 5.55125C6.25249 5.55125 7.1737 5.26249 9.27747 6.67874C11.0774 6.18372 12.9775 6.18372 14.7774 6.67874C16.8811 5.24875 17.8024 5.55125 17.8024 5.55125C18.2094 6.47334 18.2488 7.51609 17.9124 8.46623C18.6542 9.27013 19.0579 10.3288 19.0399 11.4225C19.0399 15.6436 16.4686 16.5787 14.0212 16.8537C14.2837 17.1197 14.4859 17.4392 14.614 17.7903C14.7421 18.1415 14.7932 18.516 14.7637 18.8887C14.7637 20.3599 14.7499 21.5424 14.7499 21.9136C14.7499 22.2024 14.9562 22.5461 15.5062 22.4361C17.9797 21.6033 20.0767 19.9185 21.4229 17.6825C22.7692 15.4466 23.2769 12.8049 22.8556 10.2292C22.4343 7.65346 21.1114 5.31128 19.1229 3.62074C17.1345 1.93021 14.6099 1.00136 12 1V1Z\\" fill=\\"currentColor\\"></path></svg><!----></div></a> <div class=\\"inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 leading-none text-center border border-ink-100 cursor py-1.5 px-3 text-sm bg-ink-200 has-tooltip\\" data-original-title=\\"null\\"><!---->Business<!----></div></span></h2>
    </div>
  </div>
</div>"
`;
